tests/test/org/commcare/backend/session/test/SessionNavigatorTests.java

```diff
@@ -144,5 +144,39 @@ public void testAutoSelectEnabledWithOneCaseAndNoDetail() {
         triggerSessionStepAndCheckResultCode(SessionNavigator.START_FORM_ENTRY);
     }
 
+    @Test
+    public void testSettingAndGettingSessionExtras() {
```

Comment:
This is a pretty standard session traversal. Let me know if there are corner cases that I should add to the test.